airflow/www/static/js/grid/renderTaskRows.jsx

```diff
@@ -56,36 +57,41 @@ const renderTaskRows = ({
 
 const TaskInstances = ({
   task, dagRunIds, selectedRunId, onSelect,
-}) => (
-  <Flex justifyContent="flex-end">
-    {dagRunIds.map((runId) => {
-      // Check if an instance exists for the run, or return an empty box
-      const instance = task.instances.find((gi) => gi.runId === runId);
-      const isSelected = selectedRunId === runId;
-      return (
-        <Box
-          py="4px"
-          px={boxPaddingPx}
-          className={`js-${runId}`}
-          data-selected={isSelected}
-          transition="background-color 0.2s"
-          key={`${runId}-${task.id}`}
-          bg={isSelected && 'blue.100'}
-        >
-          {instance
-            ? (
-              <StatusBox
-                instance={instance}
-                group={task}
-                onSelect={onSelect}
-              />
-            )
-            : <Box width={boxSizePx} data-testid="blank-task" />}
-        </Box>
-      );
-    })}
-  </Flex>
-);
+}) => {
+  const { filters: { taskState } } = useFilters();
```

Comment:
Calling the `useFilters` hook directly here is causing extra rerenders. We should move the hook to the parent component and pass `taskState` as a prop

Reply:
We are now using a state at higher level. Performances seem better.